pcsx2-qt/Debugger/DisassemblyView.cpp

```diff
@@ -106,6 +107,44 @@ void DisassemblyView::contextCopyInstructionText()
 	QGuiApplication::clipboard()->setText(FetchSelectionInfo(SelectionInfo::INSTRUCTIONTEXT));
 }
 
+void DisassemblyView::contextPasteInstructionText()
+{
+	if (!cpu().isCpuPaused())
+	{
+		QMessageBox::warning(this, tr("Assemble Error"), tr("Unable to change assembly while core is running"));
+		return;
+	}
+
+	// split text in clipboard by new lines
+	QString clipboardText = QApplication::clipboard()->text();
+	std::vector<std::string> newInstructions = StringUtil::splitOnNewLine(clipboardText.toLocal8Bit().constData());
+	int newInstructionsSize = newInstructions.size();
+
+	// validate new instructions before pasting them
+	for (int instructionIdx = 0; instructionIdx < newInstructionsSize; instructionIdx++)
+	{
+		u32 replaceAddress = m_selectedAddressStart + instructionIdx * 4;
+		u32 encodedInstruction;
+		std::string errorText;
+		bool valid = MipsAssembleOpcode(newInstructions[instructionIdx].c_str(), &cpu(), replaceAddress, encodedInstruction, errorText);
+		if (!valid)
+		{
+			QMessageBox::warning(this, tr("Assemble Error"), QString("%1 %2").arg(errorText.c_str()).arg(newInstructions[instructionIdx].c_str()));
+			return;
+		}
+	}
+
+	// paste validated instructions
+	for (int instructionIdx = 0; instructionIdx < newInstructionsSize; instructionIdx++)
+	{
+		u32 replaceAddress = m_selectedAddressStart + instructionIdx * 4;
+		u32 encodedInstruction;
+		std::string errorText;
+		MipsAssembleOpcode(newInstructions[instructionIdx].c_str(), &cpu(), replaceAddress, encodedInstruction, errorText);
```

Comment:
I'd favor calling MipsAssembleOpcode once here and storing the result, instead of calling it twice, since it calls into the expression parser which in turn reads from the symbols database, so there's a very slight chance of a race condition if symbols are updated between these calls.